File: presentation/src/main/java/ru/andrey/cleandictionary/presentation/view/WordListView.java
package ru.andrey.cleandictionary.presentation.view;

import android.app.Activity;

import com.arellomobile.mvp.MvpView;
import com.arellomobile.mvp.viewstate.strategy.OneExecutionStateStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;

import java.util.List;

import ru.andrey.cleandictionary.presentation.dto.TranslationDto;

@StateStrategyType(OneExecutionStateStrategy.class)
public interface WordListView extends MvpView {

    void setFavoriteMenuIcon(boolean activate);

    void startActivity(Class<? extends Activity> classActivity);

    void show(List<TranslationDto> items);
}

File: presentation/src/main/java/ru/andrey/cleandictionary/presentation/view/list/WordListView.kt
package ru.andrey.cleandictionary.presentation.view.list

import com.arellomobile.mvp.MvpView
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy
import com.arellomobile.mvp.viewstate.strategy.OneExecutionStateStrategy
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType
import ru.andrey.cleandictionary.presentation.dto.TranslationDto

@StateStrategyType(AddToEndSingleStrategy::class)
interface WordListView : MvpView {

    fun setFavoriteMenuIcon(activate: Boolean)

    @StateStrategyType(OneExecutionStateStrategy::class)
    fun openAddWord()

    fun show(items: List<TranslationDto>)

    @StateStrategyType(OneExecutionStateStrategy::class)
    fun showSnackBar()
}
